hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -402,6 +416,66 @@ public void initialize(URI name, Configuration originalConf)
       long authDirTtl = conf.getLong(METADATASTORE_AUTHORITATIVE_DIR_TTL,
           DEFAULT_METADATASTORE_AUTHORITATIVE_DIR_TTL);
       ttlTimeProvider = new S3Guard.TtlTimeProvider(authDirTtl);
+
+      S3Downloader rawS3Downloader = new S3Downloader() {
+        @Override
+        public AbortableInputStream download(
+            String requestBucket, String key, long rangeStart, long rangeEnd,
+            ChangeTracker changeTracker, String operation) throws IOException {
+          String serverSideEncryptionKey = getServerSideEncryptionKey(
+              requestBucket, getConf());
+          GetObjectRequest request = new GetObjectRequest(requestBucket, key);
+          if (S3AEncryptionMethods.SSE_C
+              .equals(getServerSideEncryptionAlgorithm()) &&
+              StringUtils.isNotBlank(serverSideEncryptionKey)) {
+            request.setSSECustomerKey(
+                new SSECustomerKey(serverSideEncryptionKey));
+          }
+
+          changeTracker.maybeApplyConstraint(request);
+
+          String text = String.format("%s %s at %d",
+              operation, uri, rangeStart);
+
+          S3Object object = Invoker.once(text, "s3a://" + key + "/" + bucket,
```

Comment:
Do the `key` and `bucket` need to be switched here? Also, use `requestBucket` instead of `bucket`?